### PR TITLE
PYMODM-95 Validating EmbeddedMongoModel in EmbeddedDocument type fields

### PR DESCRIPTION
**Pull request by shreybatra**
Checked and validated type of related_field in EmbeddedDocumentField and EmbeddedDocumentListField. Raised Value error the same was as RelatedModelFieldsBase.

### CODE REVIEW

**Comment by shreybatra** [comment]
@ShaneHarvey done the changes. Please review. :)

File: pymodm/base/fields.py [hunk 1]
```diff
@@ -267,6 +267,37 @@ def _model_to_document(self, value):
             '%s is not a valid %s' % (value, self.related_model.__name__))
 
 
+class RelatedEmbeddedModelFieldsBase(RelatedModelFieldsBase):
+    '''Base class for EmbeddedDocument and EmbeddedDocumentListField.'''
```
Comment (ShaneHarvey) [inline pymodm/base/fields.py]: Can you change to double quotes (`"""..."""`)?

File: test/test_related_fields.py [hunk 2]
```diff
@@ -194,3 +194,22 @@ def test_coerce_reference_type(self):
         comment = Comment(body='this is a comment', post=post_id).save()
         comment.refresh_from_db()
         self.assertEqual('this is a post', comment.post.body)
+
```
Comment (ShaneHarvey) [inline test/test_related_fields.py]: Thanks!

File: pymodm/base/fields.py [hunk 3]
```diff
@@ -267,6 +267,37 @@ def _model_to_document(self, value):
...
+
+    def __init__(self, model, verbose_name=None, mongo_name=None, **kwargs):
+        super(RelatedEmbeddedModelFieldsBase, self).__init__(model=model,
```
Comment (ShaneHarvey) [inline pymodm/base/fields.py]: Please wrap all lines at 80 characters.

File: pymodm/fields.py [hunk 4]
```diff
@@ -47,15 +56,6 @@
...
 
-from pymodm import validators
-from pymodm.base.fields import RelatedModelFieldsBase, GeoJSONField
-from pymodm.common import _import, validate_mongo_keys
-from pymodm.compat import text_type, string_types, PY3
-from pymodm.connection import _get_db
-from pymodm.errors import ValidationError, ConfigurationError
-from pymodm.base.fields import MongoBaseField
-from pymodm.files import File, GridFSStorage, FieldFile, ImageFieldFile
-from pymodm.vendor import parse_datetime
```
[the remark this answers is not in the document]
Comment (shreybatra) [reply pymodm/fields.py]: @ShaneHarvey done this.

File: pymodm/base/fields.py (hunk 1 above)
Comment (shreybatra) [reply pymodm/base/fields.py]: @ShaneHarvey done this.

File: pymodm/base/fields.py (hunk 3 above)
Comment (shreybatra) [reply pymodm/base/fields.py]: @ShaneHarvey done this.

**Comment by ShaneHarvey** [comment]
Thanks @shreybatra!

**Comment by shreybatra** [comment]
@ShaneHarvey ur welcome!

**Pull request merged**